src/function/algebra/simplify/util.js

```diff
@@ -14,56 +14,109 @@ export const createUtil = /* #__PURE__ */ factory(name, dependencies, ({ Functio
   // e.g. multiply is not commutative for matrices
   // The properties should be calculated from an argument to simplify, or possibly something in math.config
   // the other option is for typed() to specify a return type so that we can evaluate the type of arguments
-  const commutative = {
-    add: true,
-    multiply: true
+
+  /* So that properties of an operator fit on one line: */
+  const T = true
+  const F = false
+
+  const defaultName = 'defaultF'
+  const defaultContext = {
+    /*      */ add: { trivial: T, total: T, commutative: T, associative: T },
+    /**/ unaryPlus: { trivial: T, total: T, commutative: T, associative: T },
+    /* */ subtract: { trivial: F, total: T, commutative: F, associative: F },
+    /* */ multiply: { trivial: T, total: T, commutative: T, associative: T },
+    /*   */ divide: { trivial: F, total: T, commutative: F, associative: F },
+    /*    */ paren: { trivial: T, total: T, commutative: T, associative: F },
+    /* */ defaultF: { trivial: F, total: T, commutative: F, associative: F }
   }
-  const associative = {
-    add: true,
-    multiply: true
+  const realContext = { divide: { total: F }, log: { total: F } }
+  const positiveContext = {
+    subtract: { total: F },
+    abs: { trivial: T },
+    log: { total: T }
   }
 
-  function isCommutative (node, context) {
-    if (!isOperatorNode(node)) {
-      return true
+  function hasProperty (nodeOrName, property, context = defaultContext) {
+    let name = defaultName
+    if (typeof nodeOrName === 'string') {
+      name = nodeOrName
+    } else if (isOperatorNode(nodeOrName)) {
+      name = nodeOrName.fn.toString()
+    } else if (isFunctionNode(nodeOrName)) {
+      name = nodeOrName.name
+    } else if (isParenthesisNode(nodeOrName)) {
+      name = 'paren'
+    }
+    if (hasOwnProperty(context, name)) {
+      const properties = context[name]
+      if (hasOwnProperty(properties, property)) {
+        return properties[property]
+      }
+      if (hasOwnProperty(defaultContext, name)) {
+        return defaultContext[name][property]
+      }
+    }
+    if (hasOwnProperty(context, defaultName)) {
+      const properties = context[defaultName]
+      if (hasOwnProperty(properties, property)) {
+        return properties[property]
+      }
+      return defaultContext[defaultName][property]
     }
-    const name = node.fn.toString()
-    if (context && hasOwnProperty(context, name) && hasOwnProperty(context[name], 'commutative')) {
-      return context[name].commutative
+    /* name not found in context and context has no global default */
+    /* So use default context. */
+    if (hasOwnProperty(defaultContext, name)) {
+      const properties = defaultContext[name]
+      if (hasOwnProperty(properties, property)) {
+        return properties[property]
+      }
     }
-    return commutative[name] || false
+    return defaultContext[defaultName][property]
```

Comment:
Can it happen that we would return `undefined` here? Is there need for a fallback `... || false`?

Reply:
Since the default context has a defaultF property which in turn has all of trivial, total, commutative, and associative properties, `hasProperty` can never return `undefined` on any of these properties. But if you call `hasProperty(name, 'transcendental', defaultContext)` you will get `undefined`. But this is arguably a feature, especially since undefined is false-y: when you ask for a property, you can get true or false on known properties, and if you just check the truth value all unknown properties will be false, but if you want to see if it's a property that mathjs knows nothing about, you can specifically check for undefined. This could also potentially help a little with misspelled properties, like 'assocative'.

On the other hand, if you simply want all unknown properties to come back as the exact value `false`, I am happy to add two `|| false` guards in the spots where undefined might come back.

Reply:
So I have left this as is for now; just let me know if you want the `|| false` put in and I will do it, or feel free to do it yourself if you like.

Reply:
Good point. I like the current approach of returning `undefined` in case of an unknown function, let's go with that 👍